### PR TITLE
TFAC-329 Adding Reauthentication Logic with Redirect

### PR DESCRIPTION
**Pull request by jedtan**
I think this should do what is desired, since the onAuthProcessCompleted callback sets the redirect URL for the page.

### CODE REVIEW

File: web/src/js/components/Authentication/Authentication.js [hunk 1]
```diff
@@ -228,6 +228,11 @@ class Authentication extends React.Component {
           // authentication.
           // https://github.com/gladly-team/tab/issues/589
           this.props.fetchUser()
+
+          const urlParams = getUrlParameters()
+          if (urlParams.reauth === 'true') {
+            await this.onAuthProcessCompleted()
```
Comment (jedtan) [inline web/src/js/components/Authentication/Authentication.js:234]: Would like thoughts if I should decompose `onAuthProcessCompleted` into just a `redirect` function.

Comment (kmjennison) [reply web/src/js/components/Authentication/Authentication.js:240]: @jedtan Could you explain what you mean here? If you'd just like to move it from a method into its own function—sure!

Comment (jedtan) [reply web/src/js/components/Authentication/Authentication.js:240]: That question makes no sense when i reread it haha. I actually think my original question is moot - i was just thinking of splitting out a separate redirect function that could be called from both `onAuthProcessCompleted` and here (replacing this `onAuthProcessCompleted` call). But now that I think about it, better to reuse the same generic "do this after auth is done" logic here.

**Review by kmjennison — APPROVED** [review]
I manually tested, and it works great. LGTM! 🍝

**Pull request merged**